docs/contributor/draft/rate-limit-design.md

```diff
@@ -0,0 +1,80 @@
+# Rate Limit Proposal
+
+## Spec 
+
+| field                                            | type        | description                                                     | required                      |
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| wokloadSelector                                  | object      | when skipped, it's applied for the whole namespace workloads, if empty and in istio-system ns, applies to all workloads in all namespaces | no                            |
+| workloadSelector.labels                          | object      | allows to select given workload by a label selector                   | no                            |
+| local                                   | object      | allows to describe local rate limit properties                  | yes                           |
+| local.bucket                            | object      | allows to set rate limit capacity through bucket concept        | yes                           |
+| local.bucket.max_tokens                 | int         | max tokens that can be gathered for the rate limit capacity     | yes                           |
+| local.bucket.tokens_per_fill            | int         | number of tokens added per fill cycle                           | yes                           |
+| local.bucket.fill_interval              | time        | interval on which tokens are filled                             | yes                           |
```

Comment:
We should also have a consequence that we have validation for the minimum fill_interval value. Documentation about this can be found here: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto#local-rate-limit-proto
```suggestion
| local.bucket.fill_interval              | time        | The fill interval that tokens are added to the bucket. During each fill interval `tokens_per_fill` are added to the bucket. The bucket will never contain more than `max_tokens` tokens. The `fill_interval` must be >= 50ms to avoid too aggressive refills.                            | yes                           |
```